MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+        None,
+    )
+    .context("Failed to create PSBT")?
+    .psbt;
+let psbt = bitcoind
+    .wallet_process_psbt(&psbt, None, None, None)
+    .with_context(|| "Failed to process PSBT")?
+    .psbt;
+let psbt = load_psbt_from_base64(psbt.as_bytes()) // SHOULD BE PROVIDED BY CRATE AS HELPER USING rust-bitcoin base64 feature
+    .with_context(|| "Failed to load PSBT from base64")?;
+log::debug!("Original psbt: {:#?}", psbt);
+let pj_params = payjoin::sender::Configuration::with_fee_contribution(
+    payjoin::bitcoin::Amount::from_sat(10000),
+    None,
+);
+```
+
+### 3. (optional) Spawn a thread or async task that will broadcast the transaction after one minute unless canceled
+
+I wrote this in the original docs, but I think it should be amended.
+
+In case the payjoin goes through but you still want to pay by default. This missing `payjoin-client`
+
+Writing this, I think of [Signal's contributing guidelines](https://github.com/signalapp/Signal-iOS/blob/main/CONTRIBUTING.md#development-ideology):
+> The answer is not more options. If you feel compelled to add a preference that's exposed to the user, it's very possible you've made a wrong turn somewhere.
+
+### 4. Construct the request with the PSBT and parameters
+
+```rs
+let (req, ctx) = link
+    .create_pj_request(psbt, pj_params)
+    .with_context(|| "Failed to create payjoin request")?;
+```
+
+### 5. Send the request and receive response
+
+Senders request a payjoin from the receiver with a payload containing the Original PSBT  and optional parameters. They require a secure endpoint for authentication and message secrecy to prevent that transaction from being modified by a malicious third party during transit or being snooped on. Only https and .onion endpoints are spec-compatible payjoin endpoints.
+
+Avoiding the secure endpoint requirement is convenient for testing.
+
+```rs
+let client = reqwest::blocking::Client::builder()
+    .danger_accept_invalid_certs(danger_accept_invalid_certs)
+    .build()
+    .with_context(|| "Failed to build reqwest http client")?;
+let response = client
+    .post(req.url)
+    .body(req.body)
+    .header("Content-Type", "text/plain")
+    .send()
+    .with_context(|| "HTTP request failed")?;
+```
+
+### 6. Process the response
+
+An `Ok` response should include a Payjoin Proposal PSBT. Check that it's signed, following protocol, not trying to steal or otherwise error.
+
+```rs
+// TODO display well-known errors and log::debug the rest
+let psbt = ctx.process_response(response).with_context(|| "Failed to process response")?;
+log::debug!("Proposed psbt: {:#?}", psbt);
+```
+
+Payjoin response errors (called [receiver's errors in spec](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#receivers-well-known-errors)) come from a remote server and can be used to "maliciously to phish a non technical user." Only those listed as "well known" in the spec should be displayed with preset messages to prevent phishing.
```

Comment:
Errors from `process_response` may be improved by being separated into safe `ReceiverError::WellKnown` standard error types that can be displayed in the UI and `ReceiverError::DebugOnly` which "can only appear in debug logs." Separation would simplify an integration's error handling. (https://github.com/payjoin/rust-payjoin/issues/53)